### ISSUE TITLE
Runtime Error: Algorithm took longer than 10 minutes on a single time loop.

### ISSUE THREAD

**Issue by xenu256** [QuantConnect/Lean#144]
On algo "Check if there always data?"

Runs fine, at the end stuck with runtime errror:

  at QuantConnect.Isolator.ExecuteWithTimeLimit (TimeSpan timeSpan, System.Func`1 withinCustomLimits, System.Action codeBlock, Int64 memoryCap) [0x00000] in :0 
  at QuantConnect.Lean.Engine.Engine.Run (QuantConnect.Packets.AlgorithmNodePacket job, System.String assemblyPath) [0x00000] in Lean.Engine.Engine.Run (QuantConnect.Packets.AlgorithmNodePacket job, System.String assemblyPath) [0x00000] in :0

**Comment by bizcad** [comment]
I am getting the same error on my algo as well.  It runs fine on my machine with the latest compiled version of Lean in Debug mode.

**Comment by jaredbroad** [comment]
Thanks guys -- where are you seeing the error? On QuantConnect.com or the
open source project?

If the open source project can you please tell me your local version number?

Thanks
JB

On Wed, Jul 22, 2015 at 11:21 AM, Nicholas Stein <notifications@github.com>
wrote:

> I am getting the same error on my algo as well. It runs fine on my machine
> with the latest compiled version of Lean in Debug mode.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/QuantConnect/Lean/issues/144#issuecomment-123756576>.
>



-- 
Jared Broad

www.quantconnect.com
Phone-USA: +1 917 327 0556
*Democratizing Finance, Empowering Individuals*
Facebook <http://www.facebook.com/QuantConnect> | Twitter
<https://twitter.com/#!/QuantConnect> | LinkedIn
<http://www.linkedin.com/company/quantconnect> | Skype: jaredbroad

**Comment by xenu256** [comment]
QuantConnect.

**Comment by bizcad** [comment]
I am getting it on QuantConnect.com.  It is happening at 26% on my Abhi Goel Algorithm during backtest.
The algorithm is a custom data algorithm where QC is executing the orders.  Please feel free to take a look at it if you want.  Here is a link to the log.

http://data.quantconnect.com/backtests/7828/103038/a95620c70ff3701671b60f254c807edd-log.txt

In open source, the algorithm consistently runs well.  When I port to QuantConnect.com, The only difference is removing my custom logging; which goes to a local disk.

**Comment by jaredbroad** [comment]
Cool thank you guys, we're about to deploy a new build of QC with quite a
few bug fixes, so we might have already fixed the error.

I think the bug you're talking about has already been fixed -- MarketOnOpen
orders were causing the system to hang if you submitted an order when there
was no more data,

JB

On Wed, Jul 22, 2015 at 4:35 PM, Nicholas Stein <notifications@github.com>
wrote:

> I am getting it on QuantConnect.com. It is happening at 26% on my Abhi
> Goel Algorithm during backtest.
> The algorithm is a custom data algorithm where QC is executing the orders.
> Please feel free to take a look at it if you want. Here is a link to the
> log.
>
>
> http://data.quantconnect.com/backtests/7828/103038/a95620c70ff3701671b60f254c807edd-log.txt
>
> In open source, the algorithm consistently runs well. When I port to
> QuantConnect.com, The only difference is removing my custom logging; which
> goes to a local disk.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/QuantConnect/Lean/issues/144#issuecomment-123854214>.
>



-- 
Jared Broad

**Comment by bizcad** [comment]
I just tried the algo again, hoping you would have do the deployment by now.  I get a runtime error within a couple of minutes.  That is progress.  However, there are no logs or diagnostic information about what the runtime error is.

**Comment by mchandschuh** [comment]
@bizcad - I just took a peak at your algo, it looks like after 7 minutes the data feed fails to load your custom data subscription. A few questions, how large is the file (I tried going to the URL but got two empty lines) and have you tested the parse logic?  The logs show after 7 minutes the following:
`20150728 13:52:48 Trace:: FileSystemDataFeed.Run(): Failed to load subscription: ABHIGOAL`
Which means we weren't able to get a single data point from the remote file that had a valid timestamp.

I noticed in your custom data implementation you swallow all the errors in the try/catch. I would recommend not swallowing and just throw those errors. If you have a title row, detect the row (maybe by first letter) and return null or a default instance for that line, but you may be swallowing parse errors as well.

**Comment by bizcad** [comment]
@mchandschuh Thanks for testing.  I discovered the link to get the list of symbols is no longer working either.  
<code>        private const string DATA_SOURCE_URI = @"http://54.69.46.205/filter_csv_withcache.php?sdate=1/1/2010&edate=7/6/2015&knowhen=31,41&rd_pct=83,92,100&pid=9";
</code>
I have contacted Abhi to find out what is wrong.

As to swallowing Exceptions ins the catch block, I was using the sample code which recommended doing nothing in the catch block.  Should I remove the try catch then?  I am asking for the data without headers, so when there is no data returned, I return the object as null.  Then in BaseDataSubscritionFactory.Read line is "" and Read return EndOfStream, so SubscritionDataReader.MoveNext returns false and there is no Event.  This behavior is correct IMHO.

Again thanks for your assistance.

**Comment by mchandschuh** [comment]
If you know your data has no headers then I would recommend removing the try/catch. I'd rather exceptions be thrown then go unnoticed. I agree that the behavior is correct. We iterated all of the dates within the backtest trying to get valid custom data and just came up empty.

My suggestion was thinking that maybe you were getting a parse error and it was being swallowed (and hence not reported) via the try/catch in your Reader method.  When you get the data back online, try running again and let me know how you make out. I want to make sure whatever issue you've been observing has been resolved. We're getting ready for another deploy shortly, so hopefully is a fixed bug already.

**Comment by jaredbroad** [comment]
Closing because original bug fixed.

**Comment by bizcad** [comment]
I contacted Abhi and he told me he has to rework all his dates (whatever that means), and it will take a week or so.  When he gets back on line I will fire off the algo again.  Meanwhile I will remove the try catch  block.

**Comment by michaelsrew** [comment]
Hi, i am getting the same issue.  
I tried the latest master with 10 min bars only (TradeBarConsolidator) in debug mode without using "public void OnData(TradeBars data)". After 2 days it stopped working with the following Error:
**Security.ExecuteWithTimeLimit(): Algorithm took longer than 10 minutes on a single time loop.**
Maybe someone can help me with the problem. I attached the last part of the log.
[log.txt](https://github.com/QuantConnect/Lean/files/541554/log.txt)
thanks
